Could we remove the `Copy` requirement of `ActualT` type parameter in `Matcher` trait?

In 471d4a2, the `ActualT` changed from an associated type to a type parameter. I think it is a good change, but is `ActualT: Copy` still required? In #323, there was some discussions about lifetime problems, but I don't think that is still valid now because the lifetime of the object being matched is not tied to the matcher anymore.

The comment in the code mentioned:

```rust
// `ActualT` requires `Copy` so that `actual` could be passed to `matches` and
// if it fails passed to `explain_match`. We can relax this constraint later by
// requiring only `Clone`.
```

But `Copy` in rust is generally considered cheap (if object is so large that memcpy is expensive, then the type probably shouldn't implement Copy in the first place), so there is no harm to take `actual: &ActualT` instead. If `ActualT` is `Copy`, the implementer can dereference it by themselves. This simplify a lot of lifetime problems when implementing custom matchers.

I've been trying to migrate from 0.11 to 0.13, and I stumbled on this new requirement pretty much throughout all my code, as my assertions were built all with owned values. For example, I have a looot of assertion methods that looks like these:

```
pub fn is_output_with_failure(
    code: u16,
    message: impl Into<String>,
) -> impl Matcher<ActualT = OutputEntryMessage> {
    pat!(OutputEntryMessage {
        result: some(pat!(output_entry_message::Result::Failure(eq(
            messages::Failure {
                code: code as u32,
                message: message.into(),
            }
        ))))
    })
}
```

And now I can't really implement them anymore:

```
pub fn is_output_with_failure(
    code: u16,
    message: impl Into<String>,
) ->impl for<'a> Matcher<&'a OutputCommandMessage> {
    pat!(OutputCommandMessage {
        result: some(pat!(output_command_message::Result::Failure(eq(
            &messages::Failure {
                code: code as u32,
                message: message.into(),
            }
        ))))
    })
}
```

as obviously:

```
error[E0716]: temporary value dropped while borrowed
   --> src/tests/mod.rs:159:14
    |
157 | /      pat!(OutputCommandMessage {
158 | |          result: some(pat!(output_command_message::Result::Failure(eq(
159 | |              &messages::Failure {
    | | ______________^
160 | ||                 code: code as u32,
161 | ||                 message: message.into(),
162 | ||             }
    | ||_____________^ creates a temporary value which is freed while still in use
163 | |          ))))
164 | |      })
    | |_______- opaque type requires that borrow lasts for `'static`
165 |    }
    |    - temporary value is freed at the end of this statement

```

Essentially taking ownership in the matchers of the "expected" values was super handy to create these methods, which now is not really possible anymore.

And yes I know this specific case I could nest a `pat!` matcher here, but i have more complex cases where the `pat!` matcher doesn't work because of private fields and i really wanna test `eq`.

--- a/googletest/tests/reference_type_test.rs
+++ b/googletest/tests/reference_type_test.rs
@@ -1,0 +1,51 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Debug)]
+struct ArenaHolder<'a, T: ?Sized> {
+    value: &'a T,
+}
+
+impl<'a, T: PartialEq + ?Sized> PartialEq<T> for ArenaHolder<'a, T> {
+    fn eq(&self, other: &T) -> bool {
+        self.value == other
+    }
+}
+
+struct Strukt {
+    a_field: String,
+}
+
+impl<'a> ArenaHolder<'a, Strukt> {
+    fn get_a_field(&self) -> ArenaHolder<'_, str> {
+        ArenaHolder { value: &self.value.a_field }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use googletest::prelude::*;
+
+    #[gtest]
+    fn test() {
+        let arena = [Strukt { a_field: "hello".into() }];
+
+        let holder = ArenaHolder { value: &arena[0] };
+
+        let inner_holder = holder.get_a_field();
+
+        expect_that!(inner_holder, eq("hello"));
+    }
+}

